Dockerfile

```
@@ -0,0 +1,124 @@
+###===========================================================================
+### Python ###
+FROM python:3.8.18-slim AS python
+
+# upgrade system
+RUN apt-get update \
+    # Install security updates
+    && apt-get upgrade -y
+
+    # don’t try to write .pyc files on the import of source modules
+ENV PYTHONDONTWRITEBYTECODE=1 \
+    # allow statements and log messages to immediately appear
+    PYTHONUNBUFFERED=1 \
+    APP_DOCKER_PATH=/usr/src/app \
+    VIRTUAL_ENV=/opt/venv
+
+
+###===========================================================================
+### Flask Base builder ###
+FROM python AS flask-builder-base
+RUN apt-get install -y \
+    build-essential \
+    libxml2-dev \
+    libxslt-dev \
+    python3-venv
+
+# set builder environment variables
+ENV PIP_DEFAULT_TIMEOUT=100 \
+    # disable a pip version check to reduce run-time & log-spam
+    PIP_DISABLE_PIP_VERSION_CHECK=1 \
+    # cache is useless in docker image, so disable to reduce image size
+    PIP_NO_CACHE_DIR=1 \
+    # virtual environment path
+    VIRTUAL_ENV=/opt/venv
```

```suggestion
    PIP_NO_CACHE_DIR=1
```
`ENV` are inherited so `VIRTUAL_ENV` is already defined in python base image.